test/plugins/WebOfTrust/util/TickerDelayedBackgroundJobTest.java

```diff
@@ -0,0 +1,393 @@
+                    }
+                    try {
+                        Thread.sleep(1, rand.addAndGet(500));
+                    } catch (InterruptedException e) {
+                        wasInterrupted.set(true);
+                    }
+                }
+            }
+        };
+    }
+
+    /**
+     * Creates a Runnable that invokes {@code job.triggerExecution(long)} 1 time, sleeps for about
+     * 1 ms, and repeats this until all delays are used (first to last).
+     * The created Runnable is stateless and can be used multiple times, even concurrently.
+     */
+    private Runnable newHammerCustom(final DelayedBackgroundJob job, final long[] delays) {
+        return new Runnable() {
+            @Override
+            public void run() {
+                for (long delay : delays) {
+                    job.triggerExecution(delay);
+                    try {
+                        Thread.sleep(1, rand.addAndGet(500));
+                    } catch (InterruptedException e) {
+                        wasInterrupted.set(true);
+                    }
+                }
+            }
+        };
+    }
+
+    /**
+     * Warmup the job/ticker/executor by triggering its immediate execution and waiting for the
+     * change to happen 10 times. Restores the {@link #value} after warmup.
+     * @param job the job to warm up
+     * @param jobDuration the expected duration of the job
+     */
+    private void warmup(TickerDelayedBackgroundJob job, long jobDuration) throws Exception {
+        int val = value.getAndSet(0);
+        assertEquals(JobState.IDLE, job.getState());
+        for (int i = 1; i <= 10; i++) {
+            job.triggerExecution(0);
+            Thread.sleep(jobDuration);
+            // Wait for at most an additional 20ms for the job to finish.
+            sleeper = new Sleeper();
+            for (int j = 0; j < 20; j++) {
+                if (job.getState() == JobState.IDLE) {
+                    break;
+                }
+                sleeper.sleepUntil(j + 1);
+            }
+        }
+        assertEquals(JobState.IDLE, job.getState());
+        value.set(val);
+    }
+
+    /**
+     * Creates a new, warmed-up DelayedBackgroundJob that increments the {@link #value} by 1 and
+     * waits for {@code jobDuration} ms on each execution, with given aggregation delay.
+     * @param jobDuration the job duration
+     * @param delay the trigger aggregation delay
+     * @return
+     */
+    private TickerDelayedBackgroundJob newJob(long jobDuration, long delay, String name) throws
+            Exception {
+        Runnable test = newValueIncrementer(jobDuration);
+        TickerDelayedBackgroundJob job = new TickerDelayedBackgroundJob(test, name, delay, ticker);
+        warmup(job, jobDuration);
+        return job;
+    }
+
+    @Test
+    public void testTriggerDefault() throws Exception {
+        // First test for a reasonable fast test (with execution time smaller than the delay).
```

Comment:
The second occurrence of the word "test" in that sentence likely is a mixup, what was it supposed to be?